l1-contracts/src/governance/GSE.sol

```diff
@@ -98,27 +124,93 @@ contract GSECore is IGSECore, Ownable {
   using DelegationLib for DelegationData;
   using ProposalLib for Proposal;
 
+  // The amount of staking asset required to deposit an attester on the rollup.
   uint256 public constant DEPOSIT_AMOUNT = 100e18;
+  // The minimum amount of staking asset required to be in the set to be considered an attester.
+  // Presently, as the rollup instance does not allow for partial withdrawals, the only way for
+  // a staked amount to reduce is to either withdraw the entire amount or be slashed.
   uint256 public constant MINIMUM_STAKE = 50e18;
 
-  address public constant CANONICAL_MAGIC_ADDRESS =
-    address(uint160(uint256(keccak256("canonical"))));
+  /**
+   * Create a special "bonus" address for use by the latest rollup.
+   *
+   * As far as terminology, the GSE tracks staking and voting/delegation data for "instances",
+   * and an "instance" is either the address of a "true" rollup contract which was added via `addRollup`,
+   * or (ONLY IN THIS CONTRACT) this special "bonus" address, which has its own accounting.
+   *
+   * NB: in every other context, "instance" refers broadly to a specific instance of an aztec rollup contract
+   * (possibly inclusive of its family of related contracts e.g. Inbox, Outbox, etc.)
+   *
+   * Thus, this bonus address appears in `delegation` and `instances`, and from the perspective of the GSE,
+   * it is an instance (though it can never be in the list of rollups).
+   *
+   * Lower in the code, we use "rollup" if we know we're talking about a rollup (often msg.sender),
+   * and "instance" if we are talking about about either a rollup instance or the bonus instance.
+   *
+   * The latest rollup according to `rollups` may use the attesters and voting power
+   * from the BONUS_INSTANCE_ADDRESS as a "bonus" to their own.
+   *
+   * One invariant of the GSE is that the attesters available to any rollup instance must form a set.
+   * i.e. there must be no duplicates.
+   *
+   * Thus, for the latest rollup, there are two "buckets" of attesters available:
+   * - the attesters that are associated with the rollup's address
+   * - the attesters that are associated with the BONUS_INSTANCE_ADDRESS
+   *
+   * The GSE ensures that:
+   * - each bucket individually is a set
+   * - when you add these two buckets together, it is a set.
+   *
+   * For a rollup that is no longer the latest, the attesters available to it are the attesters that are
+   * associated with the rollup's address.
+   *
+   * In this way, the "effective" attesters/balance/etc for a rollup (at a point in time) is:
```

Comment:
As mentioned earlier, I think the implication or the "why" is good to put here. When reading if not thinking about it, people might miss that it is what enable the automatic moving. Just that we should really point it out so it is simple for anyone even without thinking to realise.